theano/gpuarray/dnn.py

```diff
@@ -1626,11 +1630,12 @@ def apply(self, fgraph):
 
 @register_opt('cudnn', 'fast_compile')
 @op_lifter([SoftmaxGrad])
-def local_softmax_dnn_grad(node, ctx_name):
+#@register_opt2([SoftmaxGrad], 'fast_compile')
+def local_softmax_dnn_grad(op, ctx_name, inputs):
```

Comment:
tags are just a way to select which optimization to use. It don't do more then that. Each optimization tell which tag will enable it. Then the optimizer will query each optimization and select the one that have tags that match the one it was requested. Adding the tag 'cudnn' allow the user to enable/disable it by using the flag optimization_including/excluding=cudnn.